mysite/announcements/views.py

```diff
@@ -0,0 +1,45 @@
+from django.shortcuts import render
+from django.contrib.auth.models import User, Group
+from django.contrib.auth import authenticate, login
+from django.http import HttpResponseRedirect
+from django import forms
+from .forms import UserRegistrationForm
+from .models import Announcements
+
+# Create your views here.
+
+
+def home(request):
+    user_id = request.user.id
+
+    current_user_groups = User.objects.filter(id=user_id)[0].groups.all()
+    valid_announcements = Announcements.objects.filter(groups__in=current_user_groups)\
+        .exclude(sent_at=None).filter(has_expired=False)
```

Comment:
Instead of one exclude and another filter.

You can have it all in one filter e.g. 

`.filter(sent_at__isnull=False, has_expired=False, groups__in=current_user_groups)`

Reply:
Noted